clientcmd/clientcmd.go

```diff
@@ -42,6 +45,31 @@ func NewClient() (*Client, error) {
 	}, nil
 }
 
+func (c *Client) PodLogs(namespace, podName string, opts *corev1.PodLogOptions) (string, error) {
+	logs, err := c.client.Pods(namespace).GetLogs(podName, opts).Stream(context.TODO())
+	if err != nil {
+		return "", err
+	}
+	defer logs.Close()
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, logs)
+	return buf.String(), err
+}
+
+func (c *Client) IsPodRunning(namespace, podName string) (bool, error) {
+	pod, err := c.client.Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	isReady := false
+	for _, v := range pod.Status.Conditions {
```

Comment:
is it works with OpenShift 3.11?
as I can see it was added in K8s 1.14 - https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate

Reply:
it should work fine on openshift 3.11
```
% oc version
Kubernetes Version: v1.11.0+d4cacc0
```
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    openshift.io/deployment-config.latest-version: "1"
    openshift.io/deployment-config.name: registry-console
    openshift.io/deployment.name: registry-console-1
    openshift.io/scc: restricted
  creationTimestamp: "2020-12-17T12:46:17Z"
  generateName: registry-console-1-
  labels:
    deployment: registry-console-1
    deploymentconfig: registry-console
    name: registry-console
  name: registry-console-1-54r95
  namespace: default
  ownerReferences:
  - apiVersion: v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicationController
    name: registry-console-1
    uid: d775a4f1-4065-11eb-97ae-02777a916ab1
  resourceVersion: "4754"
  selfLink: /api/v1/namespaces/default/pods/registry-console-1-54r95
  uid: db3c1c5c-4065-11eb-97ae-02777a916ab1
spec:
...
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2020-12-17T12:46:17Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2020-12-17T12:56:36Z"
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: null
    status: "True"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2020-12-17T12:46:17Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://aede3ea1523f4e2ded7b9eb02a8d3f46be164cf623c8db3c40b10e58ed203944
    image: registry.access.redhat.com/openshift3/registry-console:v3.11
    imageID: docker-pullable://registry.access.redhat.com/openshift3/registry-console@sha256:ec5e55c6abd8de65c1a3baac86e8e983bd38a02513e58599e3a1a27cd59468bd
    lastState:
      terminated:
        containerID: docker://f6e9851bf1373d10fcdaf3c256fc3d680f01bcf68cb09d677aeb1c0bdce64544
        exitCode: 137
        finishedAt: "2020-12-17T12:54:05Z"
        reason: Error
        startedAt: "2020-12-17T12:46:41Z"
    name: registry-console
    ready: true
    restartCount: 1
    state:
      running:
        startedAt: "2020-12-17T12:56:08Z"
  hostIP: 10.0.1.98
  phase: Running
  podIP: 10.128.0.27
  qosClass: BestEffort
  startTime: "2020-12-17T12:46:17Z"
```